### DISCUSSION TITLE
18 gennaio 2022 B - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#5] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/734e5482-67c8-406f-80dd-16038cf3a546)

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-13 17:19:57](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/d056bb81-f8ee-45e3-aa08-78285dba771c)

**Comment by atyion** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/101365532/7ce703f2-d984-40d7-878a-332ae409b2ae)

Ho un dubbio in merito all'ultimo theta join, voi che ne pensate?

**Comment by Frisbon** [comment]
Non ho capito, eggman vi premia se fate le query one-liner? 😅
1a) 
- Ricavo gli aerei revisionati <2019 
- Ricavo i voli da milano
- Ricavo i viaggi nel 2021 (da milano)
- Join tra aerei revisionati e viaggi nel 2021, prendo modello e nposti

1b)
- Ricavo volo verso NY e da NAP
- Li unisco e collego a tutti i viaggi del 2018
- Ricavo gli aerei (cattivi) che hanno fatto un volo da NAP a NY nel 2018
(per avere solo gli aerei ho proiettato solo i suoi attributi, il resto non mi permette di fare la differenza) 
- Prendo il modello dalla differenza tra tutti gli aerei - quelli che hanno fatto il volo cattivo

![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/84731341/dbfc646e-d94c-42ab-804d-3793bd64077c)

**Reply by atyion** [reply]
Slay

**Comment by Elia-Belli** [comment]
**(a)** 

- Aeroporti Milano: $Aerop\\_ MI = \rho_{(MI\leftarrow Sigla)}(\sigma_{(Città\ =\ Milano)}(Aeroporto))$
- Voli in partenza da Milano : $Voli\\_ MI = Volo \bowtie_{(Partenza\ =\ MI)} Aerop\\_ MI$
- Viaggi nel 2021: $Viaggi\\_ 2021= \sigma_{(01-01-2021\leq Data\lt 01-01-2022)}(Viaggio)$
- Voli in partenza da Milano nel 2021: $Voli\\_ MI\\_ 2021= Voli\\_ MI\bowtie_{(Sigla=SiglaVolo)}Viaggi\\_ 2021$
- Aerei Revisionati prima del 2019 : $Aerei\\_ 2019= \sigma_{(AnnoR\ <\ 2019)}(Aereo)$
- Risposta : $R=\pi_{(Modello,NPosti)}(Aerei\\_ 2019 \bowtie_{(ID\ =\ Aereo)}Voli\\_ MI\\_ 2021)$

**(b)** Calcolo tramite complemento
- Aeroporti Napoli: $Aerop\\_ NA = \rho_{(NA\leftarrow Sigla)}(\pi_{(Sigla)}(\sigma_{(Città\ =\ Napoli)}(Aeroporto)))$
- Aeroporti New York: $Aerop\\_ NY = \rho_{(NY\leftarrow Sigla)}(\pi_{(Sigla)}(\sigma_{(Città\ =\ NewYork)}(Aeroporto)))$
- Voli Napoli-NewYork: $Voli\\_ NN= (Voli\bowtie_{(Partenza\ =\ NA)}Aerop\\_ NA)\bowtie_{(Arrivo\ =\ NY)}(Aerop\\_ NY)$
- Viaggi nel 2018: $Viaggi\\_ 2018= \sigma_{(01-01-2018\leq Data\lt 01-01-2019)}(Viaggio)$
- Voli Napoli-Milano del 2018: $Voli\\_ NN\\_ 2018 = Viaggi\\_ 2018 \bowtie_{(SiglaVolo=Volo)} Voli\\_ NN$
- Aerei totali: $Aerei= \pi_{(ID)}(Aereo)$
- Aerei desiderati: $Aerei= Aerei - (\rho_{(ID\leftarrow Aereo)}((\pi_{(Aereo)}Voli\\_ NN\\_ 2018)))$
- Risposta: $\pi_{(Modello)}(Aereo\bowtie Aerei)$
